radio/src/thirdparty/Stb/stb_image.h

```diff
@@ -3829,16 +3841,25 @@ static int stbi__do_zlib(stbi__zbuf *a, char *obuf, int olen, int exp, int parse
 
 STBIDEF char *stbi_zlib_decode_malloc_guesssize(const char *buffer, int len, int initial_size, int *outlen)
 {
-   stbi__zbuf a;
+   char * r;
+   stbi__zbuf * a;
+   a = (stbi__zbuf *) stbi__malloc(sizeof(stbi__zbuf));
+   if (!a) return 0; 
    char *p = (char *) stbi__malloc(initial_size);
-   if (p == NULL) return NULL;
-   a.zbuffer = (stbi_uc *) buffer;
-   a.zbuffer_end = (stbi_uc *) buffer + len;
-   if (stbi__do_zlib(&a, p, initial_size, 1, 1)) {
-      if (outlen) *outlen = (int) (a.zout - a.zout_start);
-      return a.zout_start;
+   if (p == NULL) {
+      STBI_FREE(a);
+      return NULL;
+   }
+   a->zbuffer = (stbi_uc *) buffer;
+   a->zbuffer_end = (stbi_uc *) buffer + len;
+   if (stbi__do_zlib(a, p, initial_size, 1, 1)) {
+      if (outlen) *outlen = (int) (a->zout - a->zout_start);
+      r = a->zout_start;
+      STBI_FREE(a);
+      return r;
    } else {
-      STBI_FREE(a.zout_start);
+      STBI_FREE(a->zout_start);
```

Comment:
Yep, I was more preplexed when I saw:
```c
      r = a->zout_start;
      STBI_FREE(a);
      return r;
```
But I think it is ok, because a->zout_start is just a pointer value.